std/datetime/timezone.d

```diff
@@ -874,17 +877,23 @@ public:
       +/
     override bool dstInEffect(long stdTime) @trusted const nothrow
     {
-        import core.stdc.time : localtime, tm;
+        import core.stdc.time : tm;
+
         time_t unixTime = stdTimeToUnixTime(stdTime);
 
         version(Posix)
         {
-            tm* timeInfo = localtime(&unixTime);
+            import core.sys.posix.time : localtime_r;
+
+            tm timeInfo;
```

Comment:
`= void`?

Reply:
Is void supposed to be used whenever initialization is not required, or is it just a performance hack meant for inner loops?

Reply:
In general writing or omitting void doesn't matter, since the optimizer can determine if writing the variable is unnecessary and omit it, but here it cannot. In any case it serves to document intention.

Reply:
Yeah, it has nothing to do with inner loops specifically. It's merely a way to avoid having the variable default-initialized when you know that it's unnecessary. And in this case, there's no way that the optimizer can optimize out the default initialization, because the variable is set in a C function. Ideally, each time that this code declares a variable just to pass it to a C function to set it without looking at its value, `= void` would be used to avoid the default initialization.

Reply:
Sure... I guess.